lib/cdo/i18n_string_url_tracker.rb

```diff
@@ -58,34 +58,55 @@ def initialize
   end
 
   # Records the given string_key and URL so we can analyze later what strings are present on what pages.
-  # @param string_key [String] The key used to review the translated string from our i18n system.
   # @param url [String] The url which required the translation of the given string_key.
   # @param source [String] Context about where the string lives e.g. 'ruby', 'maze', 'turtle', etc
-  def log(string_key, url, source)
+  # @param string_key [String] The key used to locate the desired string.
+  # @param scope [Array] Array of strings representing the hierarchy leading up to the string_key.
+  # @param separator [String] The separator string used by I18n to concatenate the string_key hierarchy
+  #        into a single normalized string.
+  def log(url, source, string_key, scope = nil, separator = nil)
+    # Return if DCDO flag is unset, or we get incomplete info
     return unless DCDO.get(I18N_STRING_TRACKING_DCDO_KEY, false)
+    return unless string_key && url && source
+
     # Skip URLs we are not interested in.
     return unless allowed(url)
-
     url = normalize_url(url)
-    return unless string_key && url && source
+
+    # Scope could come in as empty or a normalized string, so make sure it's an array
+    separator ||= I18n.default_separator
+    scope = [] if scope.nil? || scope.empty?
```

Comment:
See comment about defaults above

Reply:
for scope I wanted to make sure it would get set to [] if it came in as an empty string, which the default param declaration wouldn't cover, it would just keep it as an empty string.

Reply:
actually... I think if it comes in as an empty string it will be turned into an empty array by the "if string" check below. I'll move it into the default params declaration and check that it handles an empty string as expected 👍